src/main/java/org/folio/inventory/consortium/consumers/ConsortiumInstanceSharingHandler.java

```diff
@@ -0,0 +1,273 @@
+                sendErrorResponseAndPrintLogMessage(tenantId, errorMessage, sharingInstance, record.headers());
+                promise.fail(errorMessage);
+              });
+          } else {
+            String errorMessage = format("Error checking Instance by id %s on target tenant %s. Error: %s", instanceId, sharingInstance.getTargetTenantId(), failure.getMessage());
+            sendErrorResponseAndPrintLogMessage(tenantId, errorMessage, sharingInstance, record.headers());
+            promise.fail(errorMessage);
+          }
+        })
+        .onSuccess(instanceOnTargetTenant -> {
+          String errorMessage = format("Instance %s is present on target tenant: %s", instanceId, sharingInstance.getTargetTenantId());
+          sendErrorResponseAndPrintLogMessage(tenantId, errorMessage, sharingInstance, record.headers());
+          promise.fail(errorMessage);
+        });
+      return promise.future();
+    } catch (Exception ex) {
+      LOGGER.error(format("Failed to process data import kafka record from topic %s", record.topic()), ex);
+      return Future.failedFuture(ex);
+    }
+  }
+
+  private void sendErrorResponseAndPrintLogMessage(String tenantId, String errorMessage, SharingInstance sharingInstance, List<KafkaHeader> kafkaHeaders) {
+    LOGGER.error("handle:: {}", errorMessage);
+    sendEventToKafka(tenantId, sharingInstance, ERROR, errorMessage, kafkaConfig, kafkaHeaders);
+  }
+
+  private Future<Instance> publishInstanceToTenant(Instance instance, InstanceCollection instanceCollection) {
+    JsonObject jsonInstance = instance.getJsonForStorage();
+    jsonInstance.remove("hrid");
+    return addInstance(Instance.fromJson(jsonInstance), instanceCollection);
+  }
+
+  private Future<Instance> getInstanceById(String instanceId, String tenantId, InstanceCollection instanceCollection) {
+    LOGGER.info("getInstanceById :: instanceId: {} on tenant: {}", instanceId, tenantId);
+    Promise<Instance> promise = Promise.promise();
+    instanceCollection.findById(instanceId, success -> {
+        if (success.getResult() == null) {
+          LOGGER.warn("getInstanceById :: Can't find Instance by id: {} on tenant: {}", instanceId, tenantId);
+          promise.fail(new NotFoundException(format("Can't find Instance by id: %s on tenant: %s", instanceId, tenantId)));
+        } else {
+          LOGGER.debug("getInstanceById :: Instance with id {} is present on tenant: {}", instanceId, tenantId);
+          promise.complete(success.getResult());
+        }
+      },
+      failure -> {
+        LOGGER.error(format("getInstanceById :: Error retrieving Instance by id %s on tenant %s - %s, status code %s",
+          instanceId, tenantId, failure.getReason(), failure.getStatusCode()));
+        promise.fail(failure.getReason());
+      });
+    return promise.future();
+  }
+
+  private Future<Instance> addInstance(Instance instance, InstanceCollection instanceCollection) {
```

Comment:
change the method name to "addInstanceInStorage" since you already have "updateInstanceInStorage".

Reply:
Thanks, It will be fixed in scope of [MODINV-870](https://issues.folio.org/browse/MODINV-870)